### DISCUSSION TITLE
Is it possible that key=z to 'Clear the List' only clears flows which are finished, and not ones which are in transit?

### DISCUSSION THREAD

**Discussion by Dima-369** [mitmproxy/mitmproxy#6128] [Q&A]
I like to clear the list often to see only specific flows, but often I accidentally wipe out flows which are in transit. The API call is then terminated and never passed to the client who initiated the API call.

So if my flow list shows 3 flows:

- finished
- in transit
- finished

Hitting z (or maybe a custom command?) should then only show 1 flow afterwards:

- in transit

Can this maybe be implemented via an addon?

**Comment by mhils** [comment]
Assuming you are using mitmproxy and not mitmweb, `z` is currently bound to `view.flows.remove @all`. You can probably add a custom keybinding with a different filter, see https://docs.mitmproxy.org/stable/concepts-commands/#custom-key-bindings and https://docs.mitmproxy.org/stable/concepts-filters/. :)

**Reply by Dima-369** [reply]
Great! Thanks for the resources.

I can achieve my wanted behavior like this: `view.flows.remove !~q`

**Reply by Dima-369** [reply]
And then bind it like this in `keys.yaml`:

```yaml
- key: "z"
  ctx: ["flowlist"]
  cmd: view.flows.remove !~q
```
